src/keys.rs

```diff
@@ -280,8 +275,11 @@ impl IssuanceAuthorizingKey {
             ChildIndex::try_from(coin_type)?,
             ChildIndex::try_from(account)?,
         ];
-        ExtendedSpendingKey::from_path(seed, path, ZIP32_ORCHARD_PERSONALIZATION_FOR_ISSUANCE)
-            .map(|esk| esk.sk().into())
+        let esk =
+            ExtendedSpendingKey::from_path(seed, path, ZIP32_ORCHARD_PERSONALIZATION_FOR_ISSUANCE)?;
+
+        IssuanceAuthorizingKey::from_bytes(*esk.sk().to_bytes())
+            .ok_or(zip32::Error::InvalidSpendingKey)
```

Comment:
good but let's go with
```rust
        // we are reusing zip32 logic for deriving the key, zip32 should be updated as discussed
        let &isk_bytes =
            ExtendedSpendingKey::from_path(seed, path, ZIP32_ORCHARD_PERSONALIZATION_FOR_ISSUANCE)?
                .sk()
                .to_bytes();

        IssuanceAuthorizingKey::from_bytes(isk_bytes)
            .ok_or(zip32::Error::InvalidSpendingKey)
```

Reply:
I see, makes for better readability this way. Done.